source/sigv4.c

```diff
@@ -436,6 +561,402 @@ static SigV4Status_t parseDate( const char * pDate,
 
 /*-----------------------------------------------------------*/
 
+/* Hex digest of provided parameter string. */
+static void hexEncode( SigV4String_t * pInputStr,
+                       SigV4String_t * pHexOutput )
+{
+    static const char digitArr[] = "0123456789abcdef";
+    char * hex = NULL;
+    size_t i = 0U;
+
+    assert( pInputStr != NULL );
+    assert( pHexOutput != NULL );
+    assert( pInputStr->pData != NULL );
+    assert( pHexOutput->pData != NULL );
+
+    hex = pHexOutput->pData;
+
+    for( i = 0; i < pInputStr->dataLen; i++ )
+    {
+        *( hex++ ) = digitArr[ ( pInputStr->pData[ i ] & 0xF0 ) >> 4 ];
+        *( hex++ ) = digitArr[ ( pInputStr->pData[ i ] & 0xF0 ) ];
+    }
+
+    pHexOutput->dataLen = pInputStr->dataLen * 2;
+}
+
+/*-----------------------------------------------------------*/
+
+static SigV4Status_t getCredentialScope( SigV4Parameters_t * pSigV4Params,
+                                         SigV4String_t * pCredScope )
+{
+    SigV4Status_t returnVal = SigV4InvalidParameter;
+    char * pBufWrite = NULL;
+    int32_t bytesWritten = 0;
+
+    assert( pSigV4Params != NULL );
+    assert( pCredScope != NULL );
+    assert( pCredScope->pData != NULL );
+
+    pBufWrite = pCredScope->pData;
+
+    /* Use only the first 8 characters from the provided ISO 8601 string (YYYYMMDD). */
+    bytesWritten = snprintf( ( char * ) pBufWrite,
+                             ISO_DATE_SCOPE_LEN + 1,
+                             "%*s",
+                             ISO_DATE_SCOPE_LEN,
+                             pSigV4Params->pDateIso8601 );
+
+    if( bytesWritten == ISO_DATE_SCOPE_LEN )
+    {
+        bytesWritten = snprintf( ( char * ) pBufWrite,
```

Comment:
`pBufWrite` is set to `pCredScope->pData;` - won't multiple `snprintf` calls overwrite the data or am I missing something?